helix-term/src/commands.rs

```diff
@@ -1710,8 +1710,9 @@ fn delete_selection_impl(cx: &mut Context, op: Operation) {
         let values: Vec<String> = selection.fragments(text).map(Cow::into_owned).collect();
         let reg_name = cx.register.unwrap_or('"');
         let registers = &mut cx.editor.registers;
-        let reg = registers.get_mut(reg_name);
-        reg.write(values);
+        if let Some(reg) = registers.get_mut(reg_name) {
```

Comment:
The check on line 1708 is now not necessary, but i left it there, because otherwise calculating values to store is unnecessary, and delete/change is often used as noyank version (at least i use them a lot)